control/rebalance.py

```diff
@@ -40,8 +42,9 @@ def auto_rebalance_task(self ):
                         self.logger.debug(f"Nothing found for rebalance, break at {i} iteration")
                         break
                 except Exception:
-                        self.logger.exception(f"Exception in auto rebalance")
-                        raise
+                    self.logger.exception(f"Exception in auto rebalance")
+                    parent_pid = os.getppid()
+                    os.kill(parent_pid, signal.SIGTERM)
```

Comment:
Consider raising SystemExit with an appropriate message to enable a graceful shutdown and ensure the gateway exits with a non-zero code.
```python
raise SystemExit(f"auto rebalance failed, quitting gateway")
```